### ISSUE TITLE
Exempt concatenated string literals for `black --preview` compatibility

### ISSUE THREAD

**Issue by andersk** [robsdedude/flake8-picky-parentheses#32]
[`black --preview`](https://black.readthedocs.io/en/stable/the_black_code_style/future_style.html) enforces parentheses for concatenated string literals within comma-separated expressions:

```python
items = [
    "item",
    "item",
    (
        "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt"
        " ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation"
        " ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in"
        " reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur"
        " sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id"
        " est laborum."
    ),
    "item",
]
```

These extra parentheses make it much easier to catch bugs where you’re missing commas in a list of strings, but flake8-picky-parentheses flags them as redundant with `PAR001`. Consider exempting parens around a concatenated string literal followed by a comma.

**Comment by robsdedude** [comment]
Hi and thanks for bringing this up.

I agree that these parenthesis are helpful and am inclined to add an exception for them. The exact rules when to exempt them from the lint and when not to need some thinking on my side (and playing around with black to see the black authors' option on this) .

I'm curious: why would you use a linter like flake8 in conjunction with black? Isn't black supposed to reformat the code, (almost) no questions asked? So why would one want to run a linter after that?

**Comment by andersk** [comment]
Linters go much farther than surface-level formatting! See for example the [list of rules](https://beta.ruff.rs/docs/rules/) supported by [Ruff](https://beta.ruff.rs/docs/) (a faster Rust-based successor to Flake8 plus dozens of plugins). They cover a huge spectrum of issues from performance to readability to security, and sometimes help teach you things about obscure corners of Python that you didn’t know you didn’t know.

As for your particular plugin, I used it to find a subtle bug in Zulip that had slipped past Black, Ruff, mypy, and human review:

- zulip/zulip#26710

So I’d love to help improve the logic to eliminate false positives, and then get it ported to Ruff so we can use it in our CI.

**Comment by robsdedude** [comment]
Has been released in version 0.5.1. Thanks again for the input.
